yjit/src/backend/ir.rs

```diff
@@ -77,6 +77,8 @@ pub enum Opnd
         num_bits: u8,
         /// ctx.stack_size when this operand is made. Used with idx for Opnd::Reg.
         stack_size: u8,
+        /// The number of local variables in the current ISEQ. Used only for locals.
+        local_size: Option<u32>,
```

Comment:
Would `num_locals` be a better name here?

Also, why is it an option? Can it just be 0 when there are no locals?

Reply:
> Would num_locals be a better name here?

Can you elaborate on why it's a better name? `local_size` is a variable name used throughout the CRuby codebase for this purpose. Since there are stack temps that have `stack_size`, it also seems consistent across YJIT to use `local_size` for locals.

> Also, why is it an option? Can it just be 0 when there are no locals?

At first, I thought there are `Assembler` usages where locals and `spill_temps()` are not used, so in that sense it could be optional and I wanted to distinguish ISEQs with no locals and the number not being set.

I ended up using it in outlined code like exits and stubs as well, so maybe I ended up using it everywhere. I'll see if I can always set it and remove `Option`.

Reply:
I guess I feel that `num_locals` is more descriptive given that it's the number of locals. `local_size` makes me think of a size in bytes or something like that. I wish we could go change the name in the rest of the Ruby codebase though I understand this is probably not possible. If you want to keep the same name for consistency I understand.

Reply:
The same argument applies to `stack_size`. Does `stack_size` also seem like the number of bytes to you?

Anyway, I agree `num_locals` is more descriptive about what it is, and we seem to already call it `num_locals` in a few places in YJIT, so I just renamed `local_size` to `num_locals`. https://github.com/ruby/ruby/pull/11157/commits/ba1f92609e34a5f1f21d96e708c2c2d607ac6600